packages/code-signing-certificates/src/index.ts

```diff
@@ -0,0 +1,249 @@
+  return PKI.publicKeyFromPem(publicKeyPEM);
+}
+
+/**
+ * Convert a PEM-formatted RSA private key to a private key for use with this library.
+ * @param privateKeyPEM PEM formatted private key
+ * @returns RSA private key
+ */
+export function convertPrivateKeyPEMToPrivateKey(privateKeyPEM: string): PKI.rsa.PrivateKey {
+  return PKI.privateKeyFromPem(privateKeyPEM);
+}
+
+/**
+ * Convert a PEM-formatted X.509 certificate to a certificate for use with this library.
+ * @param certificatePEM PEM formatted X.509 certificate
+ * @returns  X.509 Certificate
+ */
+export function convertCertificatePEMToCertificate(certificatePEM: string): PKI.Certificate {
+  return PKI.certificateFromPem(certificatePEM);
+}
+
+type GenerateParameters = {
+  /**
+   * Public/private key pair generated via {@link generateKeyPair}.
+   */
+  keyPair: PKI.rsa.KeyPair;
+
+  /**
+   * Certificate validity range start.
+   */
+  validityNotBefore: Date;
+
+  /**
+   * Certificate validity range end.
+   */
+  validityNotAfter: Date;
+
+  /**
+   * CN issuer and subject Distinguished Name (DN).
+   * Used for both issuer and subject in the case of self-signed certificates.
+   */
+  commonName: string;
+};
+
+/**
+ * Generate a self-signed code-signing certificate for use with expo-updates.
+ * Note that while certificate chains may be supported at some point in expo-updates, for now
+ * only self-signed certificates are supported.
+ *
+ * @returns PKI.Certificate valid for expo-updates code signing
+ */
+export function generateSelfSignedCodeSigningCertificate({
+  keyPair: { publicKey, privateKey },
+  validityNotBefore,
+  validityNotAfter,
+  commonName,
+}: GenerateParameters): PKI.Certificate {
+  const cert = PKI.createCertificate();
+  cert.publicKey = publicKey;
+  cert.serialNumber = toPositiveHex(util.bytesToHex(random.getBytesSync(9)));
+
+  assert(
+    validityNotAfter > validityNotBefore,
+    'validityNotAfter must be later than validityNotBefore'
+  );
+  cert.validity.notBefore = validityNotBefore;
+  cert.validity.notAfter = validityNotAfter;
+
+  const attrs = [
+    {
+      name: 'commonName',
+      value: commonName,
+    },
+  ];
+  cert.setSubject(attrs);
+  cert.setIssuer(attrs);
+
+  cert.setExtensions([
+    {
+      name: 'keyUsage',
+      critical: true,
+      keyCertSign: false,
+      digitalSignature: true,
+      nonRepudiation: false,
+      keyEncipherment: false,
+      dataEncipherment: false,
+    },
+    {
+      name: 'extKeyUsage',
+      critical: true,
+      serverAuth: false,
+      clientAuth: false,
+      codeSigning: true,
+      emailProtection: false,
+      timeStamping: false,
+    },
+  ]);
+
+  cert.sign(privateKey, md.sha256.create());
+  return cert;
+}
+
+function arePublicKeysEqual(key1: PKI.rsa.PublicKey, key2: PKI.rsa.PublicKey): boolean {
+  // typedef for BigInteger doesn't contain equals method
+  return (key1.n as any).equals(key2.n) && (key1.e as any).equals(key2.e);
```

Comment:
Why is `as any` necessary?

Reply:
The same question is for other `as any` in the file.

Reply:
The other way to address this is to add something to ts-declarations that adds equals() to jsbn.BigInteger's definition (maybe).